core/src/main/java/com/datastax/oss/driver/internal/core/DefaultConsistencyLevelRegistry.java

```diff
@@ -18,26 +18,36 @@
 import com.datastax.oss.driver.api.core.ConsistencyLevel;
 import com.datastax.oss.driver.api.core.DefaultConsistencyLevel;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableMap;
 import net.jcip.annotations.ThreadSafe;
 
 @ThreadSafe
 public class DefaultConsistencyLevelRegistry implements ConsistencyLevelRegistry {
 
-  private static final ImmutableList<ConsistencyLevel> values =
+  private static final ImmutableList<ConsistencyLevel> VALUES =
       ImmutableList.<ConsistencyLevel>builder().add(DefaultConsistencyLevel.values()).build();
+  private static final ImmutableMap<String, Integer> NAME_TO_CODE;
+
+  static {
+    ImmutableMap.Builder<String, Integer> nameToCodeBuilder = ImmutableMap.builder();
+    for (DefaultConsistencyLevel consistencyLevel : DefaultConsistencyLevel.values()) {
+      nameToCodeBuilder.put(consistencyLevel.name(), consistencyLevel.getProtocolCode());
+    }
+    NAME_TO_CODE = nameToCodeBuilder.build();
+  }
 
   @Override
   public ConsistencyLevel fromCode(int code) {
     return DefaultConsistencyLevel.fromCode(code);
   }
 
   @Override
-  public ConsistencyLevel fromName(String name) {
-    return DefaultConsistencyLevel.valueOf(name);
+  public int toCode(String name) {
+    return NAME_TO_CODE.get(name);
```

Comment:
This is the most significant change in this PR.
For the record I've tried immutable map, hash map and an explicit switch statement in the method, no discernable difference between the 3.

Reply:
The method is only called from `Conversions`, and (String => int) is what we really need there. Previously it would convert to the enum and immediately call `getProtocolCode()` on it, which is an unnecessary extra step (String => ConsistencyLevel => int).

As for the names I see what you mean, this is confusing so maybe we should try to be more explicit:
```java
  ConsistencyLevel codeToLevel(int code);
  int nameTocode(String name);
```

Reply:
I see, we are really trying to squeeze up to the last drop of performance here. OK then for the method names suggested in your last comment above.